lms/models/application_instance.py

```diff
@@ -190,6 +184,25 @@ def update_lms_data(self, params: dict):
 
             setattr(self, attr, params.get(attr))
 
+    def check_guid_aligns(self, tool_consumer_instance_guid):
```

Comment:
Yeah this is more explicit :+1: 

We could eventually move this to the constructor of `LTILaunchResource` and remove it from `update_lms_data`. We avoid the double call there and we'll have deeplinking (which also calls `update_lms_data` and basic_lti_lauches do the same thing.

I can live without that, not sure if it's better.